compiler/plugins/target/AMD-AIE/aie/AIEDialect.cpp

```diff
@@ -494,10 +494,10 @@ LogicalResult DMABDOp::verify() {
                << "exceeds memref size "
                << std::to_string(bufferType.getNumElements());
       }
-      if (dim.getSize() >= (1UL << 9) + 1) {
-        return emitOpError() << "Size may not exceed 1023.";
+      if (dim.getSize() >= ((1UL << 10) - 1)) {
```

Comment:
Without this fix, a convolution lit test actually fails.

Reply:
Great find!